app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java

```diff
@@ -1232,6 +1232,12 @@ public void onAuthorViewClicked() {
         if (media == null || media.getUser() == null) {
             return;
         }
+        if (sessionManager.getUserName() == null) {
+            String userProfileLink = "https://commons.wikimedia.org/wiki/User:" + media.getAuthor();
```

Comment:
Just a note on how the artist names and user names differ: Let's say I create an account with user name `RP1234` and can enter my first and last name. The artist name in this case would be `first name + last name`. So, the user's profile can be located on Commons using `RP1234` but not with `first name + last name`. This is where `getAuthor() ` would fail in the first scenario.

Reply:
Would the solution then would be just changing `media.getAuthor()` for `media.getUser()` if I understood correctly?